docs/instruments/LORISLogicParser.md

```diff
@@ -0,0 +1,143 @@
+See [Jison documentation](jison.org) for grammar and Flex pattern matching specifications.
+
+After your changes are made run `jison jison/logicParser.jison` and replace `js/logicParser.js` with the output file.
+
+### PHP Grammar Specifics
+Tokens are defined in `lexer.php`. Precedence and assertions are defined by the order of parsing functions in `parser.php`. The grammar is defined in `lexer.php`. Similarly to JS, an array defining `tag`, `op`, and `args` is output to `evaluator.php`.
+
+### Evaluator Changes
+To add new types of operations, add a case to the switch statement in `js/Evaluator.js` and `php/evaluator.php`.
+
+### Unit Testing
+JS tests can be added to `Loris/test/js-tests/Parser.test.js`. Run tests with `npm run tests:unit:js:watch`.
+
+PHP tests can be added to `Loris/test/unittests/ParserTest.php`. Run tests with `Loris/vendor/bin/phpunit --configuration phpunit.xml --testsuite 'PHPParserTest'`.
+
+# Syntax
+Note that all whitespace (spaces or tabs) is ignored in the parser.
+
+### Value Inputs
+| Type              	| Syntax                 	| Notes                                              	|
+|-------------------	|------------------------	|----------------------------------------------------	|
+| number            	| 1<br>900<br>123.456          	|                                                    	|
+| text              	| "this is my text!_123" 	| empty text is supported<br>' can be used instead of " 	|
+| variable          	| [my_variable_name]<br>[nested_variable(array_access)]<br>[my_var][nested_var]|                                                |
+| nested expression 	| (expression)           	|                                                    	|
+
+### Constants
+| Constant           	| Syntax 	|
+|--------------------	|--------	|
+| null               	| null   	|
+| true               	| true   	|
+| false              	| false  	|
+| Euler's number (e) 	| E      	|
+| pi                 	| PI     	|
+
+### Numerical Operations
+| Operation 	| Syntax 	| Notes                                                             	|
+|-----------	|--------	|--------------------------------------------------------------------   |
+| add       	| a + b  	|                                                                   	|
+| subtract  	| a - b  	|                                                                    	|
+| negate    	| - a    	|                                                                    	|
+| multiply  	| a * b  	|                                                                     	|
+| divide    	| a / b  	| cannot divide by 0                                                	|
+| exponent  	| a ^ b  	|                                                                     	|
+| percentage  	| a %   	| divides the value of a by 100                                         |
```

Comment:
why does a '%' suffix divide a value by 100?

Reply:
@driusan im not too sure. this might be an implementation from redcap. @christinerogers do you know anything from CAP?

Reply:
hold on, this makes sense . if you wanted to e.g. evaluate 5% of 20, you would write (5%*20) which would calculate to (5/100*20). whats the issue?

Reply:
Hey @zaliqarosli this operator isn't from Redcap, I'm 99% sure Zain added it as part of a greater set of functions/operators he thought we might someday use in LORIS instruments.